src/diagnostic.rs

```diff
@@ -44,67 +49,85 @@ pub struct LintDiagnostic {
   /// URL to the lint rule documentation. By default, the url uses the
   /// code to link to lint.deno.land
   pub custom_docs_url: Option<String>,
-  /// Additional information displayed beside the highlighted range.
-  pub range_description: Option<String>,
   /// Displays additional information at the end of a diagnostic.
   pub info: Vec<Cow<'static, str>>,
 }
 
+#[derive(Clone)]
+pub struct LintDiagnostic {
+  pub specifier: ModuleSpecifier,
+  /// Optional range within the file.
+  ///
+  /// Diagnostics that don't have a range mean there's something wrong with
+  /// the whole file.
+  pub range: Option<LintDiagnosticRange>,
+  pub details: LintDiagnosticDetails,
+}
```

Comment:
Split this up so a `context.add_diagnostic_details(...)` could be added that doesn't require providing a specifier.